src/commands/apply-chain-updates.ts

```diff
@@ -0,0 +1,356 @@
+import { Contract, Interface } from 'ethers'
+import { readFile } from 'fs/promises'
+import { createInterface } from 'readline/promises'
+import { PublicKey } from '@solana/web3.js'
+import TokenPoolABI from '../abi/TokenPool.ts'
+import { bigIntReplacer, chainIdFromName, getProviderNetwork } from '../lib/utils.ts'
+import type { Providers } from '../providers.ts'
+import { Format } from './types.ts'
+import { getWallet } from './utils.ts'
+import type { TypedContract } from 'ethers-abitype'
+
+type ChainToAdd = {
+  remoteChainSelector: bigint
+  remotePoolAddresses: string[]
+  remoteTokenAddress: string
+  outboundRateLimiterConfig: {
+    isEnabled: boolean
+    capacity: bigint
+    rate: bigint
+  }
+  inboundRateLimiterConfig: {
+    isEnabled: boolean
+    capacity: bigint
+    rate: bigint
+  }
+}
+
+/**
+ * Detects if an address looks like a Solana address (Base58 format, 32-44 chars)
+ */
+function looksLikeSolanaAddress(address: string): boolean {
+  // Solana addresses are Base58 and typically 32-44 characters
+  const SOLANA_ADDRESS_REGEX = /^[1-9A-HJ-NP-Za-km-z]{32,44}$/
+  return SOLANA_ADDRESS_REGEX.test(address)
+}
+
+/**
+ * Best-effort Solana address encoder: if the input looks like and is a valid Solana address,
+ * returns a 32-byte 0x-prefixed hex string; otherwise returns the original input.
+ * Only processes addresses that pass the Solana format check to avoid noise.
+ */
+function tryEncodeSolanaAddressToBytes32(address: string): string {
```

Comment:
is there a type alias for 0xstring that we should used instead of `string` return type? if not no worries